### PR TITLE
[ISSUE #314]Fix the RCE caused by the vulnerability in YAML deseriali…

### PR DESCRIPTION
**Pull request by stone-98**
## What is the purpose of the change

see: https://bitbucket.org/snakeyaml/snakeyaml/issues/561/cve-2022-1471-vulnerability-in#comment-64581479
related: https://github.com/nacos-group/nacos-spring-project/issues/314

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: nacos-spring-context/src/main/java/com/alibaba/nacos/spring/util/parse/DefaultYamlConfigParse.java [hunk 1]
```diff
@@ -49,189 +49,152 @@
...
-					return delegate.entrySet();
-				}
-			};
-		}
-	}
-
+    
+    protected static final Logger LOGGER = LoggerFactory.getLogger(DefaultYamlConfigParse.class);
+    
+    private static final String YAML_ALLOW_COMPLEX_OBJECT = "yamlAllowComplexObject";
```
Comment (stone-98) [inline nacos-spring-context/src/main/java/com/alibaba/nacos/spring/util/parse/DefaultYamlConfigParse.java:55]: I've noticed that similar configurations use camelCase naming conventions, such as ignoreResourceNotFound and ignoreUnresolvablePlaceholders, so I've adopted camelCase naming for consistency.

**Comment by stone-98** [comment]
I have a question: Why doesn't the nacos-spring-project module use com.alibaba.nacos.client.env.NacosClientProperties to retrieve properties?

File: README.md [hunk 2]
```diff
@@ -315,8 +316,33 @@ The `UserNacosConfigConverter` class binds the `@NacosConfigListener.converter()
     }
 ```
 
+##### 4.1.2.1.1. Type Conversion during YAML Parsing
 
+By default, this library uses `SafeConstructor` for type conversion during YAML parsing. This is done to ensure that potentially unsafe code is not executed during the parsing process. `SafeConstructor` provides a secure construction logic for mapping YAML structures to Java objects.
```
Comment (KomachiSion) [inline README.md]: 需要添加起始版本的说明，比如1.1.1版本前不是，1.1.2版本开始默认使用SafeConstructor

File: README.md [hunk 3]
```diff
@@ -315,8 +316,33 @@ The `UserNacosConfigConverter` class binds the `@NacosConfigListener.converter()
...
+**If You Must Use `Constructor`**
+
+- Ensure that the source of the YAML data is secure.
+
+**How to Enable `SafeConstructor`**
+
+You can set the toggle by adding a JVM system property when starting your application. For example, in the command line:
+
+```bash
+java -DyamlAllowComplexObject=true -jar your-application.jar
```
Comment (KomachiSion) [inline README.md:344]: Safe不是默认开启的吗？

这里应该是Disable吧？

Comment (stone-98) [reply README.md:344]: done

File: README.md (hunk 2 above)
Comment (stone-98) [reply README.md]: done

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**